src/NuGet.Core/NuGet.Commands/RestoreCommand/ContentFiles/VirtualFileProvider.cs

```diff
@@ -16,59 +16,55 @@ namespace NuGet.Commands
     internal class VirtualFileProvider : IFileProvider
     {
         public const string RootDir = "ROOT";
-        private readonly List<string[]> _files;
+        private readonly string _originalFile;
+        private string[] _splitFile;
         private const string ForwardSlash = "/";
         private const char ForwardSlashChar = '/';
 
-        public VirtualFileProvider(List<string> files)
+        public VirtualFileProvider(string file)
         {
-            _files = files.Select(file => file.Split(ForwardSlashChar)).ToList();
+            _originalFile = file;
+            _splitFile = null;
         }
 
         public IDirectoryContents GetDirectoryContents(string subpath)
         {
-            var contents = new List<IFileInfo>();
             var subPathParts = subpath.Split(ForwardSlashChar);
 
             // Remove the root identifier from the relative path
-            if (string.Equals(subPathParts.FirstOrDefault(), RootDir, StringComparison.Ordinal))
+            var subPathPartsOffset = subPathParts.Length > 0 && string.Equals(subPathParts[0], RootDir, StringComparison.Ordinal) ? 1 : 0;
+            _splitFile ??= _originalFile.Split(ForwardSlashChar);
+
+            int i = 0;
+            // Walk the path as long as both the file and subpath contain the same directories
+            while (i < _splitFile.Length - 1
+                && i + subPathPartsOffset < subPathParts.Length
+                && string.Equals(_splitFile[i], subPathParts[i + subPathPartsOffset], StringComparison.OrdinalIgnoreCase))
```

Comment:
ctrl-f for `_splitFile[` only finds this copy, which means that it's never assigned to, this `string.Equals` will always return false, and `.Take` elsewhere is always going to return nulls.

Am I reading the PR wrong?